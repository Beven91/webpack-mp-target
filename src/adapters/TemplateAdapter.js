module.exports = class CssAdapter {
  constructor(from, target) {
    this.from = from;
    this.target = target;
    this.fromExt = from.template;
    this.targetExt = target.template;
    this.replaceRegex = new RegExp('src=.+?' + this.fromExt.replace('.', '\.') + '(\'|")', 'ig');
  }

  apply(asset) {
    let source = String(asset.source());
    // 处理@import 值类的后缀名
    source = source.replace(this.replaceRegex, (a, b) => {
      return a.replace(this.fromExt + b, this.targetExt + b);
    });
    asset.source = () => source;
    asset.size = () => source.length;
  }
}