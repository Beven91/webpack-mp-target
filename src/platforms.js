module.exports = {
  'wx': {
    template: '.wxml',
    css: '.wxss',
    global: 'wx',
  },
  'alipay': {
    template: '.axml',
    css: '.axss',
    global: 'a'
  },
  'bytedance': {
    template: '.ttml',
    css: '.ttss',
    global: 'tt'
  }
}