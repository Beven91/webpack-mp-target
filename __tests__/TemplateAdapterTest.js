/**
 * @module TemplateAdapter
 * @name 测试 TemplateAdapter模块
 * @description
 */

import TemplateAdapter from '../src/adapters/TemplateAdapter';
import platforms from '../src/platforms';

describe('TemplateAdapter', () => {
  test('apply.bytedance', () => {

    // 创建一个从微信小安程序  转换到 字节跳动小程序的适配器
    const adapter = new TemplateAdapter(platforms.wx, platforms.bytedance);

    const asset = {
      source: () => {
        return `
        <view>
          <template src="a/b/c.wxml"/>
          <template src="a/b/c.wxml/>
          <view>111</view>
          <template src='c/d/e.wxml'></template><template src="c/d/hello.wxml"></template>
        </view>
    `
      }
    }

    // 执行转换处理
    adapter.apply(asset);
    // 断言: 匹配转换镜像
    expect(asset.source()).toMatchSnapshot();
    // 断言: 长度匹配
    expect(asset.size()).toMatchSnapshot();
  });
});
