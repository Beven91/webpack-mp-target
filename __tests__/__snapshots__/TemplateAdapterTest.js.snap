// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateAdapter apply.bytedance 1`] = `
"
        <view>
          <template src=\\"a/b/c.ttml\\"/>
          <template src=\\"a/b/c.wxml/>
          <view>111</view>
          <template src='c/d/e.ttml'></template><template src=\\"c/d/hello.ttml\\"></template>
        </view>
    "
`;

exports[`TemplateAdapter apply.bytedance 2`] = `231`;
